File: first-quarter/introduction-to-programming/namespaces-and-scope/namespaces-and-scope.cpp
// This file is an example of how namespaces and scope work in C++.
#include <bits/stdc++.h>

// This is the global namespace and scope. Everything declared here is accessible from anywhere in the program.
int x = 10;

int main() {
    // This is the main function's scope. Everything declared here is accessible from anywhere inside the main function.
    int x = 20;

    // This is the block scope. Everything declared here is accessible from anywhere inside the block.
    {
        int x = 30;
        std::cout << x << std::endl; // 30
    }

    std::cout << x << std::endl; // 20

    return 0;
}

File: first-quarter/introduction-to-programming/namespaces-and-scope/namespaces-and-scope.py
# This file is an example of how namespaces and scope work in Python.

# This is the global namespace and scope. Everything defined here is available to the entire program.
x = 10

def my_func():
    # This is a local scope. Everything declared inside this function can only be accessed by other things in the function.
    # In Python3.10, you can use the nonlocal keyword to access variables in the parent scope.
    x = 5
    print(x) # 5

my_func()

print(x) # 10
